rcl/src/rcl/expand_topic_name.c

```diff
@@ -220,15 +206,6 @@ rcl_expand_topic_name(
       return RCL_RET_BAD_ALLOC;
     }
   }
-  // if the original input_topic_name has not yet be copied into new memory, strdup it now
-  if (!local_output) {
-    local_output = rcutils_strdup(input_topic_name, allocator);
-    if (!local_output) {
-      *output_topic_name = NULL;
-      RCL_SET_ERROR_MSG("failed to allocate memory for output topic");
-      return RCL_RET_BAD_ALLOC;
-    }
-  }
```

Comment:
There's no logical path in this function that can result in the execution of this branch.